main.nf

```diff
@@ -1,17 +1,21 @@
 #!/usr/bin/env nextflow
 nextflow.enable.dsl=2 
 
-process sayHello {
+params.list_file = 'list.txt'
+
+process countLines {
   input: 
-    val x
-  output:
-    stdout
+    path list_file
   script:
+    file_content = file(list_file)
+    all_files = file_content.readLines()
+    for(each_file : all_files) {
+      println "File: ${each_file} has total of ${file(each_file).countLines()} lines "
+    }
```

Comment:
One of the advantages of Nextflow is that it can parallelize individual tasks. By including a for-loop in the `script` directive, the files will be processed sequentially. If you had 1000 files to process, it's best if that were done in parallel. 

Can you modify the Nextflow script to enable parallelism? As a tip, you will need to populate a channel with the lines in the `list_file`.

Reply:
@BrunoGrandePhD I've updated the file. I have no idea why GitHub didn't let me `Re-request` a review from you (I clicked the button but nothing happened so I'm just gonna tag you in each comment, lol